h2o-core/src/main/java/water/util/VecUtils.java

```diff
@@ -382,11 +402,56 @@ public static Vec categoricalDomainsToNumeric(final Vec src) {
       }.doAll(Vec.T_NUM, src).outputFrame().anyVec();
   }
 
+  public static class CollectDoubleDomain extends MRTask<CollectDoubleDomain> {
+    final double [] _knownDomain;
+    private IcedHashMap<IcedDouble,IcedInt> _uniques; // new uniques
+    final int _maxDomain;
+    final IcedInt _placeHolder = new IcedInt(1);
+
+    public CollectDoubleDomain(double [] knownDomain, int maxDomainSize) {
+      _maxDomain = maxDomainSize;
+      _knownDomain = knownDomain;
+    }
+
+    @Override public void setupLocal(){
+      _uniques = new IcedHashMap<>();
+    }
+
+    public double [] domain(){
+      double [] res = MemoryManager.malloc8d(_uniques.size());
+      int i = 0;
+      for(IcedDouble v:_uniques.keySet())
+        res[i++] = v._val;
+      Arrays.sort(res);
+      return _knownDomain == null?res:ArrayUtils.append(_knownDomain,res);
```

Comment:
Is the knownDomain not sorted together with the result because you want to preserve the indices?